thorlcr/graph/thgraphslave.cpp

```diff
@@ -965,10 +969,10 @@ class CThorCodeContextSlave : public CThorCodeContextBase, implements IEngineCon
         e->setOrigin(source);
         e->setAction(tea_warning);
         e->setSeverity((ErrorSeverity)severity);
-        job.fireException(e);
+        jobChannel.fireException(e);
     }
-    virtual unsigned getNodes() { return job.queryJobGroup().ordinality()-1; }
-    virtual unsigned getNodeNum() { return job.queryMyRank()-1; }
+    virtual unsigned getNodes() { return jobChannel.queryJob().queryJobGroup().ordinality()-1; }
```

Comment:
observation:  could use queryJob().querySlaves();  It would have been slightly easier to check when reviewing.